cfme/automate/dialog_element.py

```diff
@@ -0,0 +1,196 @@
+from widgetastic.xpath import quote
+
+from cfme.fixtures import pytest_selenium as sel
+from utils.appliance import Navigatable
+from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
+
+from .dialog_box import AddBoxView
+from . import AutomateCustomizationView
+
+
+class ElementForm(AddBoxView):
+    ele_label = Input(name='field_label')
+    ele_name = Input(name="field_name")
+    ele_desc = Input(name="field_description")
+    choose_type = BootstrapSelect('field_typ')
+    default_text_box = Input(name='field_default_value')
+    default_value = Checkbox(name='field_default_value')
+    field_required = Checkbox(name='field_required')
+    field_past_dates = Checkbox(name='field_past_dates')
+    field_entry_point = Input(name='field_entry_point')
+    field_show_refresh_button = Checkbox(name='field_show_refresh_button')
+    entry_value = Input(name='entry[value]')
+    entry_description = Input(name='entry[description]')
+    add_entry_button = Image('.//input[@id="accept"]')
+    field_category = BootstrapSelect('field_category')
+    text_area = Input(name='field_default_value')
+    dynamic_chkbox = Checkbox(name='field_dynamic')
+    entry_table = Table('//div[@id="field_values_div"]/form/table')
+    text_area = TextInput(id='field_default_value')
+
+    element_tree = ManageIQTree('dialog_edit_treebox')
+    dynamic_tree = ManageIQTree('automate_treebox')
+    bt_tree = ManageIQTree('automate_treebox')
+
+    apply_btn = Button('Apply')
+
+
+class AddElementView(ElementForm):
+
+    add_button = Button("Add")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and self.service_dialogs.is_opened and
+            self.title.text == "Adding a new Dialog [Element Information]"
+        )
+
+
+class EditElementView(ElementForm):
+    save_button = Button('Save')
+    reset_button = Button('Reset')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and self.service_dialogs.is_opened and
+            self.title.text == "Editing Dialog {} [Element Information]".format(self.ele_label)
+        )
+
+
+class DetailsDialogView(AutomateCustomizationView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and self.service_dialogs.is_opened and
+            self.title.text == 'Dialog "{}"'.format(self.context['object'].
+                parent.parent.parent.label)
+        )
+
+
+class ElementCollection(Navigatable):
+    def __init__(self, parent):
+        self.parent = parent
+        Navigatable.__init__(self, appliance=parent.appliance)
+
+    @property
+    def tree_path(self):
+        return self.parent.tree_path
+
+    def instantiate(self, element_data=None):
+        return Element(self,
+            element_data=element_data)
+
+    def create(self, element_data=None):
+        for each_element in element_data:
```

Comment:
**REQUIRED:** Please rename this to ```element``` and not ```each_element``` for assist in readability, it sounds like each_element is multiple elements, but in reality, it's one.